frontend/app/admin/src/views/settings/membership-types/components/PlatformMembershipTypePriceConfigEdit.vue

```diff
@@ -0,0 +1,136 @@
+                            <span class="icon trash" />
+                        </button>
+                        <button v-else-if="model.prices.size > 1" class="button text" type="button" :disabled="true">
+                            <span class="icon trash" />
+                        </button>
+                    </div>
+                </template>
+            </STListItem>
+        </STList>
+
+        <p>
+            <button class="button text" type="button" @click="addPriceForTag">
+                <span class="icon add" />
+                <span>{{ $t('admin.settings.membershipTypes.period.addPriceForTag') }}</span>
+            </button>
+        </p>
+    </div>
+</template>
+
+
+<script setup lang="ts">
+import { ComponentWithProperties, usePresent } from '@simonbackx/vue-app-navigation';
+import { DateSelection, ErrorBox, PriceInput, usePatchMap, Validator } from '@stamhoofd/components';
+import { OrganizationTag, PlatformMembershipTypeConfigPrice, ReduceablePrice } from '@stamhoofd/structures';
+import { computed } from 'vue';
+import OrganizationTagSelector from './OrganizationTagSelector.vue';
+import PlatformMembershipTypeReduceablePriceEdit from './PlatformMembershipTypeReduceablePriceEdit.vue';
+
+defineProps<{
+    hasMultiplePrices: boolean;
+    showStartDate: boolean;
+    showPricePerDay: boolean;
+    errorBox: ErrorBox | null;
+    validator: Validator;
+}>();
+
+const model = defineModel<PlatformMembershipTypeConfigPrice>({required: true});
+
+const emits = defineEmits<{(e: 'delete'): void}>();
+const present = usePresent();
+
+const {addPut: addPricePut, addDelete: addDeletePrice, patch: $pricesPatch} = usePatchMap(model.value.prices);
+
+const $startDate = computed({
+    get: () => model.value.startDate,
+    set: (startDate) => model.value = model.value.patch({startDate}),
+});
+
+const $pricePerDay = computed({
+    get: () => model.value.pricePerDay,
+    set: (pricePerDay) => model.value = model.value.patch({pricePerDay}),
+})
+
+function patchReduceablePrice(tagId: string, reduceablePrice: ReduceablePrice) {
+    addPricePut(tagId, reduceablePrice);
+    model.value = model.value.patch({prices: $pricesPatch.value})
+}
+
+async function addPriceForTag() {
+    await present({
+        modalDisplayStyle: 'popup',
+        components: [
+            new ComponentWithProperties(OrganizationTagSelector, {
+                tagIds: Array.from(model.value.prices.keys()).filter(id => id !== ''),
+                onAdd: async (_allTags: OrganizationTag[], addedTags: OrganizationTag[], deletedTags: OrganizationTag[]) => {
+                    addedTags.forEach(tag => addPriceForTagId(tag.id));
+                    deletedTags.forEach(tag => deletePriceForTagId(tag.id));
+                }
+            })
+        ]
+    });
+}
+
+function addPriceForTagId(tagId: string) {
+    addPricePut(tagId, ReduceablePrice.create({}));
+    model.value = model.value.patch({prices: $pricesPatch.value})
```

Comment:
Hier ga je eigenlijk dezelfde patch blijven applyen op hetzelfde object.

-> tag toevoegen -> je voegt dat toe in je lokale patch
-> je patcht de model met je lokale patch
-> je model verandert naar je patch
-> nog een tag toevoegen -> je voegt die toe in je lokale patch
-> je patcht de model met je lokale patch, maar dat is nu een patch met 2 tags, waarvan je al één had doorgestuurd, dus die je nodeloos nog eens patcht
-> doordat het gewoon een set is, werkt het wel, maar het kan in de toekomst verdoken bugs creeren

Reply:
Je kan dit vermijden door altijd reactive te werken, dit is niet zo een mooi patroon. Het is niet nodig om hier nog een lokale patch bij te houden via usePatchMap